src/MobileRedirection.php

```diff
@@ -78,6 +78,14 @@ public function register() {
 				add_filter( 'get_comments_link', [ $this, 'add_noamp_mobile_query_var' ] ); // For get_comments_link().
 				add_filter( 'respond_link', [ $this, 'add_noamp_mobile_query_var' ] ); // For comments_popup_link().
 			}
+
+			return;
+		}
+
+		// Show `Exit mobile version` link even mobile redirection is disabled and in Reader mode.
+		if ( AMP_Theme_Support::READER_MODE_SLUG === AMP_Options_Manager::get_option( Option::THEME_SUPPORT ) && $this->is_mobile_request() ) {
```

Comment:
Might as well make this an `else if` instead of using a `return` in the first conditional.